### PR TITLE
Add config-server package for semi-hosted onboarding

### PR DESCRIPTION
**Pull request by cyrusagent**
## Summary

This PR implements Phase 1 of the semi-hosted onboarding architecture by creating a new `@cyrus-ai/config-server` package. This package ports core update-server functionality from Go to TypeScript, enabling customers to run Cyrus locally with ngrok tunnels instead of requiring DigitalOcean droplets.

**Linear Issue**: CYPACK-185

### New Package: `@cyrus-ai/config-server`

A production-ready, open-source TypeScript package that provides an embedded HTTP server for local Cyrus configuration during onboarding.

**Components**:
- 3 Core Handler modules (ported from Go):
  - `github-handler.ts` - GitHub CLI credential management
  - `config-handler.ts` - Cyrus config.json management with backups
  - `repository-handler.ts` - Git repository cloning, deletion, listing

- Express HTTP Server:
  - Health check endpoint (`GET /health`)
  - 6 protected API endpoints under `/api/config/*`
  - Bearer token authentication middleware
  - CORS, Helmet security, and Morgan logging
  - Comprehensive error handling

- Complete Test Suite:
  - 46 tests across all handlers
  - 100% pass rate
  - Full coverage of success paths, edge cases, and error scenarios

## Review Changes Addressed

Per PR review feedback, the following changes were made:

1. ✅ **Removed default Stripe customer ID** - `stripeCustomerId` is now truly optional, only added when provided
2. ✅ **Removed default global_setup_script** - `global_setup_script` is now truly optional, only added when provided
3. ✅ **Removed env-handler.ts entirely** - Environment variable management not supported in this release
4. ✅ **Removed mcp-handler.ts entirely** - MCP configuration endpoints not supported this way

**Note**: Repository configurations can still reference MCP config files via the `mcpConfigPath` field - only the MCP management endpoints were removed.

### Porting Strategy
- Analyzed Go update-server codebase for core functionality
- Created TypeScript equivalents maintaining feature parity
- Used modern async/await patterns throughout
- Implemented proper TypeScript strict typing

### API Endpoints

**Public**:
- `GET /health` - Server health check

**Protected** (require Bearer token):
- `POST /api/config/github` - Update GitHub credentials
- `POST /api/config/cyrus-config` - Update main Cyrus configuration
- `GET /api/cyrus-config` - Retrieve current configuration
- `POST /api/config/repository` - Clone repository
- `DELETE /api/config/repository` - Delete repository + worktrees
- `GET /api/repositories` - List all repositories

### Security Features
- Bearer token authentication on all protected endpoints
- Path traversal prevention (repository names sanitized)
- Input validation on all endpoints
- File permissions set correctly (0644 files, 0755 dirs)
- Helmet security headers, CORS support

### Test Results
```
Test Files:  3 passed (3)
Tests:       46 passed (46)
Duration:    221ms
```

### Coverage
- ✅ All handlers (GitHub, config, repository)
- ✅ Success paths for all operations
- ✅ Edge cases (empty inputs, path traversal, special characters)
- ✅ Error handling scenarios
- ✅ Integration scenarios

### Build & Type Checking
- ✅ TypeScript compilation: SUCCESS (no errors)
- ✅ Type checking: PASSED
- ✅ Linting: PASSED (biome check clean)
- ✅ All tests passing

## Architecture Impact

This package enables the **semi-hosted model**:
- Customers run Cyrus locally on their machines
- No DigitalOcean droplets needed → **99% cost reduction**
- ngrok tunnel provides HTTPS access
- Vercel app sends configuration to customer's local server
- All secrets stay on customer's machine

**Cost Savings**: From $1,200-2,000/month (100 customers) to ~$12/month

## Files Changed

**New Package Structure**:
```
packages/config-server/
├── src/
│   ├── handlers/           # 3 handlers + tests (ported from Go)
│   ├── middleware/         # Auth & logging
│   ├── routes/            # Health & config routes
│   ├── types/             # Type definitions
│   ├── server.ts          # Express server
│   └── index.ts           # Main entry point
├── package.json
├── tsconfig.json
└── README.md             # Complete API documentation
```

## Verification Completed

All quality checks passed:
- ✅ 46/46 tests passing (100% pass rate)
- ✅ TypeScript type checking: zero errors
- ✅ Linting: zero issues (biome check)
- ✅ Code quality review: production-ready
- ✅ Documentation: accurate and comprehensive
- ✅ Security: proper authentication, path validation, input sanitization

## Next Steps (Not in This PR)

This PR completes **Phase 1** of the semi-hosted onboarding plan. Remaining phases:

- **Phase 2**: CLI Integration
  - Add `cyrus onboard --customer-id` command
  - Implement OnboardingService
  - Add ngrok integration
  
- **Phase 3**: Vercel App Updates
  - New API routes for CLI verification
  - Update onboarding flow
  - Webhook proxy routing

- **Phase 4**: Testing & Documentation
- **Phase 5**: Beta Launch & GA

## Breaking Changes

None. This is a new package that doesn't affect existing functionality.

## Migration Notes

Not applicable - new package, no migration needed.

### Focus Areas
- Handler logic correctness (comparison with Go originals)
- Security: authentication, path validation, input sanitization
- Test coverage and quality
- API design and RESTfulness
- Documentation completeness

### Package Ready For
- ✅ npm publication (open-source, MIT license)
- ✅ Integration into CLI (Phase 2)
- ✅ Production deployment

---

**Related**: Linear CYPACK-185

### CODE REVIEW

File: packages/config-server/src/handlers/config-handler.ts [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+	builder: ["Feature"],
+	scoper: ["PRD"],
+};
+
+/**
+ * Default values for global configuration fields
+ */
+const DEFAULT_DISALLOWED_TOOLS = ["Bash(sudo:*)"];
+const DEFAULT_NGROK_AUTH_TOKEN = "";
+const DEFAULT_STRIPE_CUSTOMER_ID = "cus_8172616126";
```
Comment (PaytonWebber) [inline packages/config-server/src/handlers/config-handler.ts]: **REMOVE THIS** there should be NO default Stripe customer ID

File: packages/config-server/src/handlers/config-handler.ts [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+const DEFAULT_MODEL = "opus";
+const DEFAULT_FALLBACK_MODEL = "sonnet";
+const DEFAULT_GLOBAL_SETUP_SCRIPT = "/opt/cyrus/scripts/global-setup.sh";
```
Comment (PaytonWebber) [inline packages/config-server/src/handlers/config-handler.ts]: **REMOVE THIS.** This script **WILL NOT** be present on the client

File: packages/config-server/src/handlers/env-handler.ts [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+		indent: 2,
+		lineWidth: 0, // Don't wrap long lines
+	});
+
+	await fs.writeFile(manifestPath, yamlData, { mode: 0o644 });
+
+	console.log(
+		`Successfully updated ${payload.variables.length} environment variables for repository ${payload.repository}, file ${payload.file_path}`,
+	);
+}
```
Comment (PaytonWebber) [inline packages/config-server/src/handlers/env-handler.ts]: **REMOVE THIS.** We currently **DO NOT** support ENV var management.

File: packages/config-server/src/handlers/mcp-handler.ts [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+ *      description: tool.description
+ *    }));
+ *    ```
+ *
+ * 4. Return response with actual server info and tools
+ *
+ * 5. Handle cleanup (close connection) in finally block
+ *
+ * See the Go implementation in test_mcp.go for reference.
+ */
```
Comment (PaytonWebber) [inline packages/config-server/src/handlers/mcp-handler.ts]: **REMOVE THIS.** We currently DO NOT support MCP configurations this way
